pkg/dns/util/util.go

```diff
@@ -119,8 +160,22 @@ func IsServiceIPSet(service *corev1.Service) bool {
 
 // GetClusterIPs returns IPs set for the service
 func GetClusterIPs(service *corev1.Service) []string {
```

Comment:
I would prefer this function return actual net.IP so the normalization issue is taken care of at the closest point of use as requiring normalization here introduces a non-local dependency in the behavior of the code.

Note that technically speaking, IPv4 also has abbreviations, e.g. "1.2" is a valid IP address, but no one knows about it so it is less common.

Reply:
My intention was to hide implementation details. If we would return `net.IP` that would leak to the consumers that normalization is done via `net.IP`.

If you feel there's a need to update the func to return `net.IP` I would ofc. But I'd prefer not to do that as for me the signature `func GetClusterIPs(service *corev1.Service) []string` is what we need - `GetClusterIPs` would return list of valid IPs of the service.